Framework/PythonInterface/mantid/api/src/Exports/ITableWorkspace.cpp

```diff
@@ -448,6 +448,34 @@ void setCell(ITableWorkspace &self, const bpl::object &col_or_row,
   getCellLoc(self, col_or_row, row_or_col, column, row);
   setValue(column, row, value);
 }
+} // namespace
+
+/**
+ * Get the contents of the workspace as a python dictionary
+ *
+ * @param self A reference to the TableWorkspace python object that we were
+ * called on
+ * @returns a boost python dictionary object with keys that are column names and
+ * values which are lists of the column values.
+ */
+bpl::dict toDict(ITableWorkspace &self) {
+  bpl::dict result;
+
+  for (const auto &name : self.getColumnNames()) {
+    const auto column = self.getColumn(name);
+    const auto &typeID = column->get_type_info();
+
+    bpl::list values;
+    for (size_t i = 0; i < self.rowCount(); ++i) {
+      bpl::handle<> handle(getValue(column, typeID, static_cast<int>(i)));
+      bpl::object obj(handle);
+      values.append(obj);
+    }
+
+    result[name] = values;
```

Comment:
Can generating the `results` dictionary be done more efficiently? For example, can we avoid a temporary `list` by using either a move assignment operator or constructing it in place? May we use a numpy array or other alternative to a `list`?